routes/blogPostsRoutes.js

```diff
@@ -0,0 +1,108 @@
+const express = require('express');
+
+const db = require('../data/db');
+
+const router = express.Router();
+
+// *************************************** POST
+
+router.post('/', async (req, res) => {
+	try {
+		const { title, contents } = req.body;
+		const newPost = { title, contents };
+		if (!title || !contents) {
+			res.status(400).json({ message: 'Please provide title and contents for the post' });
+		} else {
+			const post = await db.insert(newPost);
+			const completePost = await db.findById(post.id);
+			res.status(201).json({ success: true, completePost }); //Need to return the full post
+		}
+	} catch (error) {
+		console.log(error);
+		res.status(500).json({
+			error: 'Error retrieving the posts',
+		});
+	}
+});
+
+// *************************************** GET all posts
+
+router.get('/', async (req, res) => {
+	try {
+		const posts = await db.find();
+		res.status(200).json(posts);
+	} catch (error) {
+		console.log(error);
+		res.status(500).json({
+			error: 'The posts information could not be retrieved.',
+		});
+	}
+});
+
+// *************************************** GET post by id
+
+router.get('/:id', async (req, res) => {
+	try {
+		const { id } = req.params;
+		const post = await db.findById(id);
+		if (post.id) {
+			res.status(200).json(post);
+		} else {
+			res.status(404).json({ message: 'The post with the specified ID does not exist.' });
+		}
+	} catch (error) {
+		console.log(error);
+		res.status(500).json({
+			error: 'The post information could not be retrieved.',
+		});
+	}
+});
```

Comment:
Your GET by ID did not work properly upon testing. I think you should take a look at your function's logic once more, especially your conditionals.